sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4504,9 +4504,9 @@ def _detect_abnormal_non_terminated_nodes(
 
     def remove_cluster_config(self, handle: CloudVmRayResourceHandle) -> None:
         """Remove the YAML config of a cluster."""
+        common_utils.remove_file_if_exists(handle.cluster_yaml)
         handle.cluster_yaml = None
         global_user_state.update_cluster_handle(handle.cluster_name, handle)
```

Comment:
If interrupted between deleting the file and removing from the global_user_state, we could end up with global_user_state pointing to a non-existent file.
```suggestion
        cluster_yaml_path = handle.cluster_yaml
        handle.cluster_yaml = None
        global_user_state.update_cluster_handle(handle.cluster_name, handle)
        common_utils.remove_file_if_exists(cluster_yaml_path)
```